### ISSUE TITLE
Adding repo description with Chi/Jap charaters cause 500 error on ARMv7 platform

### ISSUE THREAD

**Issue by dragonandy** [go-gitea/gitea#4856]
- Gitea version (or commit ref):1.4.2 - 1.5.1 ARMv7
- Git version:2.7.4
- Operating system:Ubuntu 16.04 ARM Build
- Database (use `[x]`):
  - [ ] PostgreSQL
  - [x] MySQL
  - [ ] MSSQL
  - [ ] SQLite
- Can you reproduce the bug at https://try.gitea.io:
  - [ ] Yes (provide example URL)
  - [x] No
  - [ ] Not relevant
- Log gist:
[xorm.log](https://gist.github.com/dragonandy/7867cf9d5a3dd9fb169a08e96b167298)  
[gitea.log](https://gist.github.com/dragonandy/725bd87cf69635656b97e23e1963e793)

## Description

MySQL Version 5.7.23-0ubuntu0.16.04.1(Ubuntu) Linux  (armv7l)
Adding repo description with Chinese or Japanese charaters cause 500 error on ARMv7 platform
This issue seens to be platform specific only appear on my armv7 board The x86_64 version works just fine with those charaters
I've tested it on gitea 1.4.2 through 1.5.1 with my armv7 board and it all behave the same with a 500 error
The log shows its likely to be a MySQL roll back problem
I try manually change the MySQL database charset/collation to utf8-default and utf8mb4 and still get the 500 error when i try with those char in the description

## Screenshots
<img width="1157" alt="thanks1" src="https://user-images.githubusercontent.com/4011375/44970976-7f3d4380-af86-11e8-88a6-d06b2ed5d4a0.png">
<img width="1135" alt="thanks2" src="https://user-images.githubusercontent.com/4011375/44971002-9419d700-af86-11e8-93be-e43055c27936.png">

**Comment by joeyworld** [comment]
+1 here. Same issue happens also with Korean characters.

**Comment by lunny** [comment]
We need some arm devices to investigate it. Maybe a arm server? @tboerger

**Comment by tboerger** [comment]
> Maybe a arm server? @tboerger

Just rent a Scaleway machine.

**Comment by lafriks** [comment]
Could be sqlite3 problem?

**Comment by dragonandy** [comment]
> Could be sqlite3 problem?

probably not
I use a mysql database

**Comment by dragonandy** [comment]
I figure out that I can get around with this weird issue by using a database dumped from the database created by gitea x64 version and import it into my ARM device

Here is what I have done
1. I wrote some Chi/Jap characters in the repo descriptions in my x64 version of gitea for example `你好`
2. use `./gitea dump` to dump the gitea data from the x64 side
3. use mysqlworkbench import the .sql file into my ARM board with the same schema name I use for gitea then grant the same privileges etc. make sure they both have the same settings
4. I found that the default charset I imported into the ARM side of mysql is Latin so I change it into utf8_general_ci
5. restart nginx gitea mysqlserver and I can see Chinese charcters appear in the project description and I can modify them without causing a 500 error

To be clear that I use Ubuntu 16.04 on both x64 and ARM side of system
Still doubt that this is a character set problem
